app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,25 @@
 class AdminController < ApplicationController
+  before_action :restrict_non_admin_access
   def index
     @total_orders = Order.count
   end
+
+  def reports
+    if params["from"]
+      @orders = Order.where(created_at: (params["from"].to_date)..(params["to"].to_date + 1))
+    else
+      @orders = Order.where(created_at: (Date.today - 5.day)..Date.tomorrow)
+    end
+  end
+
+  def categories
```

Comment:
categories_controller#index

Reply:
created in another branch will merge it after review